persistence/src/main/java/edu/softserve/zoo/persistence/repository/impl/StatisticsRepositoryImpl.java

```diff
@@ -0,0 +1,97 @@
+package edu.softserve.zoo.persistence.repository.impl;
+
+import edu.softserve.zoo.exceptions.ApplicationException;
+import edu.softserve.zoo.exceptions.NotFoundException;
+import edu.softserve.zoo.exceptions.persistence.PersistenceException;
+import edu.softserve.zoo.model.Employee;
+import edu.softserve.zoo.model.Task.TaskStatus;
+import edu.softserve.zoo.model.Task.TaskType;
+import edu.softserve.zoo.persistence.exception.PersistenceReason;
+import edu.softserve.zoo.persistence.provider.PersistenceProvider;
+import edu.softserve.zoo.persistence.repository.StatisticsRepository;
+import edu.softserve.zoo.persistence.specification.impl.statistics.StatisticsGetEmployeeTasksStatuses;
+import edu.softserve.zoo.persistence.specification.impl.statistics.StatisticsGetEmployeeTasksTypes;
+import edu.softserve.zoo.persistence.specification.impl.statistics.StatisticsGetFedAnimalsSpecification;
+import edu.softserve.zoo.persistence.specification.impl.statistics.StatisticsGetZooStatistics;
+import edu.softserve.zoo.util.CollectionUtils;
+import edu.softserve.zoo.util.Validator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+/**
+ * @author Taras Zubrei
+ */
+@Repository
+public class StatisticsRepositoryImpl implements StatisticsRepository {
+    @Autowired
+    PersistenceProvider persistenceProvider;
+    @Override
+    @Transactional(propagation = Propagation.MANDATORY, readOnly = true)
+    public Float getFedAnimalsPercentage() {
+        return CollectionUtils.<Float>getSingleItemOrElse(persistenceProvider.find(new StatisticsGetFedAnimalsSpecification()),
```

Comment:
T findOne(Long id, Class< T > type); - it doesn't fit. I need to get aggregation function result, not some Entity by Id.